src/components/v5/common/ActionSidebar/partials/MultiSigSidebar/partials/VoteButton/VoteButton.tsx

```diff
@@ -60,56 +52,38 @@ const VoteButton: FC<VoteButtonProps> = ({
     [MultiSigVote.Reject]: MSG.reject,
   };
 
-  const transform = mapPayload(() => ({
-    colonyAddress: colony.colonyAddress,
-    colonyDomains: extractColonyDomains(colony.domains),
-    colonyRoles: extractColonyRoles(colony.roles),
-    vote: voteType,
-    domainId: multiSigDomainId,
-    multiSigId,
-    roles:
-      getRolesNeededForMultiSigAction({
-        actionType,
-        createdIn: multiSigDomainId,
-      }) || [],
-  }));
+  const getVotePayload = () => {
+    handleLoadingChange(true);
+
+    return {
+      colonyAddress: colony.colonyAddress,
+      colonyDomains: extractColonyDomains(colony.domains),
+      colonyRoles: extractColonyRoles(colony.roles),
+      vote: voteType,
+      domainId: multiSigDomainId,
+      multiSigId,
+      roles:
+        getRolesNeededForMultiSigAction({
+          actionType,
+          createdIn: multiSigDomainId,
+        }) || [],
+    };
+  };
 
   return (
-    <ActionForm
+    <ActionButton
+      isFullSize
+      useTxLoader
       actionType={ActionTypes.MULTISIG_VOTE}
-      transform={transform}
-      onSuccess={() => {
-        setExpectedStep(VoteExpectedStep.cancel);
-      }}
+      isLoading={isLoading}
       onError={() => {
-        setExpectedStep(null);
+        handleLoadingChange(false);
       }}
+      values={getVotePayload}
+      {...buttonProps}
     >
-      {({ formState: { isSubmitting } }) =>
-        isPending || isSubmitting ? (
-          <IconButton
-            rounded="s"
-            isFullSize
-            text={{ id: 'button.pending' }}
-            icon={
-              <span className="ml-2 flex shrink-0">
-                <SpinnerGap size={18} className="animate-spin" />
-              </span>
-            }
-            className="!px-4 !text-md"
-          />
-        ) : (
-          <Button
-            type="submit"
-            {...buttonProps}
-            isFullSize
-            onClick={() => setCurrentVote(voteType)}
-          >
-            {formatText(buttonText[voteType])}
-          </Button>
-        )
-      }
-    </ActionForm>
+      {formatText(buttonText[voteType])}
```

Comment:
is it any chance that buttonText[voteType] will return undefined?

Reply:
It honestly shouldn't since it's an enum of just 3 :crossed_fingers: :D